doc/man3/OSSL_HTTP_parse_url.pod

```diff
@@ -0,0 +1,74 @@
+=pod
+
+=head1 NAME
+
+OSSL_HTTP_parse_url,
+OCSP_parse_url
+- http utility functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/http.h>
+
+ int OSSL_HTTP_parse_url(const char *url,
+                         int *pssl, char **puser, char **phost,
+                         char **pport, int *pport_num,
+                         char **ppath, char **pquery, char **pfrag);
+
+Deprecated since OpenSSL 3.0, can be hidden entirely by defining
+B<OPENSSL_API_COMPAT> with a suitable version value, see
+L<openssl_user_macros(7)>:
+
+ int OCSP_parse_url(const char *url, char **phost, char **pport, char **ppath,
+                    int *pssl);
+
+=head1 DESCRIPTION
+
+OSSL_HTTP_parse_url() parses its input string I<url> as a URL of the form
+C<[http[s]://][userinfo@]host[:port][/path][?query][#fragment]> and splits it up
+into userinfo, host, port, path, query, and fragment components
+and a flag indicating whether it begins with C<https>.
+The host component may be a DNS name or an IP address
+where IPv6 addresses should be enclosed in square brackets C<[> and C<]>.
+The port component is optional and defaults to "443" for HTTPS, else "80".
+If the I<pport_num> argument is NULL the port specification can be in mnemonic
+form such as C<http> like with L<BIO_set_conn_port(3)>, else it must be in
+numerical form and its integer value is assigned to I<*pport_num> on success.
```

Comment:
https://tools.ietf.org/html/rfc3986#section-3.2.3

You're likely never to see a service name used as a port, as that's invalid by the standard.

Reply:
True that this feature of `BIO_set_conn_port()` is hardly (known and) used,
and since moreover RFC 3986 forbids it for URLs I've disabled it already for the parsing.